### PR TITLE
DU client

### PR DESCRIPTION
**Pull request by timoxley**

### CODE REVIEW

File: packages/client/src/DataUnionClient.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+export class DataUnionClient extends DataUnionClientBase {
+    constructor(options: DataUnionClientConfig = {}, parentContainer = rootContainer) {
+        const config = createStrictConfig(options)
+        const { childContainer: c } = initContainer(config, parentContainer)
+        super(
+            c,
+            c.resolve<Context>(Context as any),
+            c.resolve<Ethereum>(Ethereum),
+            c.resolve<Session>(Session),
+            c .resolve<LoginEndpoints>(LoginEndpoints),
```
Comment (kare) [inline packages/client/src/DataUnionClient.ts:135]: Space typo

Comment (timoxley) [reply packages/client/src/DataUnionClient.ts]: Good spotting

**Comment by tumppi** [comment]
Tried alpha.3 in core-frontend project and got error below. Is this a known issue? This happens after I add the import without even using it.

```
ERROR in ./node_modules/@dataunions/client/web/index.js 25608:37
Module parse failed: Unexpected token (25608:37)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|   }
|   getAllProviders() {
>     if (!this.ethereumConfig.network?.rpcs || !this.ethereumConfig.network?.rpcs?.length) {
|       return [getDefaultProvider()];
|     }
 @ ./src/marketplace/modules/dataUnion/services.js 11:0-49
 @ ./src/marketplace/containers/EditProductPage/DeployDataUnionModal.jsx
 @ ./src/userpages/components/DataUnionPage/index.jsx
 @ ./src/app/Userpages.jsx
 @ ./src/app/index.jsx
 @ ./src/index.jsx
 @ multi ./src/shared/utils/diagnostics.js ./src/setup.js ./src/index.jsx
```

File: scripts/copyAbis.sh [hunk 2]
```diff
@@ -4,9 +4,9 @@ set -ex
 # start in this /scripts directory
 cd "$(dirname "$0")"
 
-cd ../packages/data-union-solidity
+cd ../packages/contracts
 npm run build
 # jq .abi build/contracts/DataUnionFactorySidechain.json > ../dataunion-thegraph-subgraph/abis/DataUnionFactorySidechain.json
 # jq .abi build/contracts/DataUnionSidechain.json > ../dataunion-thegraph-subgraph/abis/DataUnionSidechain.json
 jq .abi artifacts/contracts/DataUnionFactorySidechain.sol/DataUnionFactorySidechain.json > ../dataunion-thegraph-subgraph/abis/DataUnionFactorySidechain.json
```
Comment (jtakalai) [inline scripts/copyAbis.sh:11]: these still have old directory names

File: scripts/generateYaml.sh [hunk 3]
```diff
@@ -1,15 +0,0 @@
-#!/bin/bash
```
Comment (jtakalai) [inline scripts/generateYaml.sh:1]: what mechanism replaced this in subgraph yaml generation? When new local deployment is done, those yaml need to be re-generated to match the new addresses...

**Review by jtakalai — CHANGES_REQUESTED** [review]
We should do a walkthrough of some of the mechanisms that come from StreamrClient, esp. the tsyringe/DI stuff

**Pull request merged**